### ISSUE TITLE
tensor.view loses gradient

### ISSUE THREAD

**Issue by leedtan** [pytorch/pytorch#32708]
## 🐛 Bug

can access gradient of tensor in model path after .backward(). However, gradient not available in nodes immediately after .view()

## To Reproduce
correct behavior:
lhs_pos = torch.linspace(
    1, 32, steps=32, requires_grad = True, dtype=torch.float)

lhs_pos2 = lhs_pos.view(
    2, 4, 2, 2
)
lhs_pos2.sum().backward()
lhs_pos.grad -> tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.])

Problematic case (probably incorrect behavior):
lhs_pos = torch.linspace(
    1, 32, steps=32, requires_grad = True, dtype=torch.float)

lhs_pos = lhs_pos.view(
    2, 4, 2, 2
)
lhs_pos.sum().backward()
lhs_pos.grad -> None

## Expected behavior
Expect lhs_pos.grad to also be a bunch of 1s.

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:

I'm not sure how to evaluate the environment inspection script from a bento notebook. However, I believe @mrshenli  knows how to recreate the issue. Please let me know if you have any questions about how to recreate.

**Comment by mrshenli** [comment]
@leedtan looking at this again, I think it is because `view` is an operator here, and when you assign it to `lhs_pos`, `lhs_pos` becomes non-leaf node which does not require grad by default. You can ask it to keep grad by calling `lhs_pos.retain_grad()` after the view operation.

**Comment by mrshenli** [comment]
Closing this issue as this is by design.
